### PR TITLE
BL-312 Feature/current course details 

### PR DESCRIPTION
**Pull request by dconley1212**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Loom Video

Please paste a link here of a quick loom video walking through what you did in your PR .

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have removed unnecessary comments/console logs from my code
- [ ] I have made corresponding changes to the documentation if necessary (optional)
- [ ] My changes generate no new warnings
- [ ] I have checked my code and corrected any misspellings
- [ ] No duplicate code left within changed files
- [ ] Size of pull request kept to a minimum
- [ ] Pull request description clearly describes changes made & motivations for said changes
https://www.loom.com/share/056f7508a4e94c8eb9e81ae88ce56908

### CODE REVIEW

**Comment by dconley1212** [comment]
https://www.loom.com/share/056f7508a4e94c8eb9e81ae88ce56908

File: api/children/childrenModel.js [hunk 1]
```diff
@@ -4,46 +4,32 @@ const getChildren = async () => {
...
-      container.instructor_id = x.instructor_id;
-      container.course_type_id = x.course_type_id;
-      container.start_time = x.start_time;
-      container.end_time = x.end_time;
-      container.start_date = x.start_date;
-      container.end_date = x.end_date;
-      container.location = x.location;
-      return container;
-    }),
+    .join('instructors', 'instructors.instructor_id', 'courses.instructor_id')
```
Comment (EduarddoEscobar) [inline api/children/childrenModel.js:66]: Good job shortening enrolledCourses

File: api/children/childrenRouter.js [hunk 2]
```diff
@@ -53,19 +53,20 @@ router.post('/', authRequired, checkChildObject, async function (
...
-  next
-) {
-  try {
-    const { id } = req.params;
-    const enrollments = await Children.getEnrolledCourses(id);
-    res.status(200).json(enrollments);
-  } catch (error) {
-    next(error);
+router.get(
+  '/:child_id/enrollments',
```
Comment (EduarddoEscobar) [inline api/children/childrenRouter.js:89]: Love the more specific endpoint!

**Comment by dconley1212** [comment]
https://www.loom.com/share/db53b21b775e439eb6d673fdd460ba56

**Review by EduarddoEscobar — APPROVED** [review]
Overall great work!

File: api/children/childrenMiddleware.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 const Children = require('./childrenModel');
 const checkChildExist = async (req, res, next) => {
-  const id = req.params.id;
+  const id = req.params.child_id;
```
Comment (joeys1122) [inline api/children/childrenMiddleware.js]: Good idea changing the id names, it's good to specify where exactly the id comes from

File: api/children/childrenModel.js [hunk 4]
```diff
@@ -4,46 +4,31 @@ const getChildren = async () => {
...
     .first();
 };
 
 const addChild = async (child) => {
   return await db('children').insert(child).returning('*');
 };
 
-const getEnrolledCourses = async (id) => {
-  // receives error if the id of the course doesn't exist exist  this have to modified later on we need to make a middleware that checks fot the existing courses then runs this model'
+const getEnrolledCourses = async (child_id) => {
```
Comment (joeys1122) [inline api/children/childrenModel.js:62]: thank you for updating this function, now the child enrollments endpoint gives back much more useful data!

**Review by joeys1122 — APPROVED** [review]
Overall great work getting these endpoints fixed!

**Review by aamay022 — COMMENTED** [review]
Great work inside the model function to retrieve more than just the children object, but the profile data as well. And great fix in changing the params, to know we are grabbing the child_id.

**Review by Berenika14 — CHANGES_REQUESTED** [review]
If you could add a short description in the PR and check all the boxes that represent the changes you made it will be easier to understand the changes without the need to watch the loom video. Overall great work on the code.

**Review by Ike888 — APPROVED** [review]
Great work fixing those linter errors.

**Review by ivanovicu — APPROVED** [review]

**Review by spencerlonghurst — APPROVED** [review]
Based on what I can see, everything looks good and solid. No comments or console.logs. Great job!

**Review by leewp8 — APPROVED** [review]
Looks like some useful updates to the codebase. You've both done a great job of writing the code so it looks well organized and easy for another dev to understand.

**Review by Ogden-R — APPROVED** [review]
These updates look great. i love that you put more details for the child signup endpoint.

**Review by ShahWahidy — APPROVED** [review]
you guys did a great job refactor the code and getting rid of redundancy. child looks much cleaner with just the spread operator instead of piling up unnecessary code. good luck with future PRs

File: api/children/childrenRouter.js [hunk 5]
```diff
@@ -85,22 +85,23 @@ router.delete(
...
+  authRequired,
+  checkChildExist,
+  async function (req, res, next) {
```
Comment (dirkknibbe) [inline api/children/childrenRouter.js:92]: I like the use of your checkChildExist middleware as well as the addition of authRequired 👍

File: api/children/childrenRouter.js [hunk 6]
```diff
@@ -85,22 +85,23 @@ router.delete(
...
+    } catch (error) {
+      next(error);
+    }
   }
-});
+);
 
 router.post(
-  '/:id/enrollments',
+  '/:child_id/enrollments',
```
Comment (dirkknibbe) [inline api/children/childrenRouter.js:104]: Nice, child_id is so much clearer

**Review by dirkknibbe — APPROVED** [review]

File: package-lock.json [hunk 7]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "labs-api-starter",
   "version": "1.0.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
```
Comment (ashtilawat) [inline package-lock.json:4]: This PR can't be merged until the package-lock is the same as the one on Main

**Review by ashtilawat — CHANGES_REQUESTED** [review]
To preserve the package-lock on Main, please copy and paste the package-lock from the main branch on the browser into your branch on VScode. You can "View Raw" and open the package-lock on the browser. Then use Cmd A to select all, Cmd C and Cmd V to copy paste.

**Review by jose-mercado — APPROVED** [review]
Great job refactoring the childrenModel, I also REALLY appreciate including a loom to the update, I think it provides more context and Isn't done enough by others! Awesome work!

**Review by ivanovicu — APPROVED** [review]

**Review by Rebecca-Stone — COMMENTED** [review]
The package-lock.json needs to be edited to merge, but everything else looks great! The code look clean and dry, great job!

**Review by laurensruiz — APPROVED** [review]
The endpoints made look good and more accessible. Updates to the model also look great and easier to understand for future devs. The code looks clean.

**Pull request closed without merging**